python/lsst/afw/image/makeVisitInfo.py

```diff
@@ -0,0 +1,93 @@
+#
+# LSST Data Management System
+# Copyright 2016 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from lsst.daf.base import DateTime
+from lsst.afw.coord import Coord, IcrsCoord, Observatory, Weather
+from lsst.afw.geom import Angle
+from .imageLib import VisitInfo, RotType_UNKNOWN
+
+__all__ = ["makeVisitInfo"]
+
+nanFloat = float("nan")
+nanAngle = Angle(nanFloat)
+
+
+def makeVisitInfo(
+    exposureId=0,
+    exposureTime=nanFloat,
+    date=DateTime(),
+    ut1=nanFloat,
+    era=nanAngle,
+    boresightRaDec=IcrsCoord(nanAngle, nanAngle),
+    boresightAzAlt=Coord(nanAngle, nanAngle),
+    boresightAirmass=nanFloat,
+    boresightRotAngle=nanAngle,
+    rotType=RotType_UNKNOWN,
+    observatory=Observatory(nanAngle, nanAngle, nanFloat),
+    weather=Weather(nanFloat, nanFloat, nanFloat),
+):
+    """Make a VisitInfo from keyword arguments
+
+    This function will be replaced by a VisitInfo constructor once we switch to pybind11
```

Comment:
pybind11 has support for wrapping existing code while providing named arguments. SWIG can apparently do it, but I don't know how. I did spend some time trying to %extend the class in SWIG to provide a keyword-based constructor, but was not successful and it's probably just as well, since it would have wrapped the same constructor that SWIG was already exposing without keyword arguments.